### PR TITLE
add fix to infinite Scroll

### PR DESCRIPTION
**Pull request by YafeiHu**

### CODE REVIEW

File: scripts/griddle.jsx [hunk 1]
```diff
@@ -689,11 +695,8 @@ var Griddle = React.createClass({
         var isFound;
 
         if(isChecked) {
-            isFound = find(selectedRowIds, function (item) {
-                return id === item;
-            });
-
-            if(isFound === undefined) {
+            isFound = selectedRowIds.indexOf(id);
```
Comment (jasonmorganson) [inline scripts/griddle.jsx:698]: Is this faster than lodash find?

File: scripts/griddle.jsx [hunk 2]
```diff
@@ -649,8 +650,13 @@ var Griddle = React.createClass({
         }
     },
     _toggleSelectAll: function () {
-        var visibleRows = this.getDataForRender(this.getCurrentResults(), this.columnSettings.getColumns(), true),
-            newIsSelectAllChecked = !this.state.isSelectAllChecked,
+        var visibleRows = [];
+        if (this.props.enableInfiniteScroll) {
+            visibleRows = this.getCurrentResults();
```
Comment (jasonmorganson) [inline scripts/griddle.jsx]: It looks like `getDataForRender` is doing a lot of stuff, can we really just drop it for infinite scroll?

File: scripts/griddle.jsx (hunk 1 above)
Comment (YafeiHu) [reply scripts/griddle.jsx:698]: Yes, about 1 sec. But I was using Timeline to compare the difference, I can write code to get a more precise result.

File: scripts/griddle.jsx (hunk 2 above)
Comment (YafeiHu) [inline scripts/griddle.jsx]: It depends on whether we need custom sort for some columns and if all cells are text or can be other components. If so then we need to change code here and add two more features to table.

File: scripts/griddle.jsx (hunk 1 above)
Comment (jasonmorganson) [reply scripts/griddle.jsx:698]: Cool

Comment (jasonmorganson) [reply scripts/griddle.jsx:698]: Very cool

**Review by jasonmorganson — APPROVED** [review]
👍

**Pull request merged**